src/write_syslog.c

```diff
@@ -0,0 +1,697 @@
+ *   Scott Sanders <scott at jssjr.com>
+ *   Pierre-Yves Ritschard <pyr at spootnik.org>
+ * Based on the write_tsdb plugin. Authors:
+ *   Brett Hawn <bhawn at llnw.com>
+ *   Kevin Bowling <kbowling@llnw.com>
+ * write_syslog. Authors:
+ *   Shirly Radco <sradco@redhat.com>
+ **/
+
+/* write_syslog plugin configuation example
+ *
+ * <Plugin write_syslog>
+ *   <Node>
+ *     Host "localhost"
+ *     Port "44514"
+ *     Prefix "collectd"
+ *     MessageFormat "human"
+ *     HostTags ""
+ *   </Node>
+ * </Plugin>
+ *
+ *
+ * Host - Hostname or address to connect to. Defaults to localhost.
+ * Port - Service name or port number to connect to. Defaults to 44514.
+ * Prefix - Sets the metrics prefix string. Mandatory. Defaults to collectd.
+ * MessageFormat - Sets the format in which messages are sent to the broker,
+ *                 human or json. Defaults to human.
+ *
+ *    Syslog message format:
+ *    <priority>VERSION ISOTIMESTAMP HOSTNAME APPLICATION PID MESSAGEID
+ * STRUCTURED-DATA MSG
+ *
+ *    The difference between the message formats are in the STRUCTURED-DATA and
+ * MSG parts.
+ *
+ *    Human format:
+ *     ```
+ *     <166>1 ISOTIMESTAMP HOSTNAME collectd PID MESSAGEID
+ *     ["collectd" "value": "v1" "plugin"="plugin_v"
+ *      "plugin_instance"="plugin_instance_v"
+ *      "type_instance"="type_instance_v" "type"="type_v" "ds_name"="ds_name_v"
+ *      "interval"="interval_v" ] "host_tag_example"="host_tag_example_v"
+ *       plugin_v.type_v.ds_name_v="v1"
+ *     ```
+ *
+ *     Json format:
+ *
+ *     ```
+ *     <166>1 ISOTIMESTAMP HOSTNAME collectd PID MESSAGEID STRUCTURED-DATA
+ *     {
+ *      "collectd": {
+ *        "time": time_as_epoch, "interval": interval_v, "plugin": "plugin_v",
+ *        "plugin_instance": "plugin_instance_v", "type":"type_v",
+ *        "type_instance": "type_instance_v", "plugin_v": {"type_v": v1}
+ *      } , "host":"host_v", "host_tag_example": "host_tag_example_v"
+ *     }
+ *     ```
+ * HostTags -When set, HostTags is added to the end of the metric.
+ *   It is intended to be used for adding additional metadata to the metric
+ * with.
+ *   Dots and whitespace are not escaped in this string.
+ *   When MessageFormat is set to "human" text should be in valid syslog
+ *   structured data format. No spaces allowed.
+ *   Example:
+ *     [ovirt engine="test"][ovirt vm="test2"]
+ *   When MessageFormat is set to "json", text should be in valid Json format.
+ *   Escaping the quotation marks is required.
+ *   Example:
+ *    HostTags "\"ovirt\": {\"engine\":\"test\",\"vm\":\"test2\"}"
+ * StoreRates false|true
+ *   If set to true, convert counter values to rates. If set to false (the
+ *   default) counter values are stored as is, as an increasing integer number.
+ */
+
+#include "collectd.h"
+#include "utils/common/common.h"
+
+#include "plugin.h"
+#include "utils_cache.h"
+#include "utils_random.h"
+
+#include <netdb.h>
+
+#ifndef WS_DEFAULT_NODE
```

Comment:
All these ifndefs are unneeded, no one will ever override them.

Reply:
can you please fix this?

Reply:
fixed.Please note that this is not fixed in write_tsdb.